Doppler.HtmlEditorApi/Repositories.DopplerDb/DopplerCampaignContentRepository.cs

```diff
@@ -59,23 +60,32 @@ public async Task<CampaignState> GetCampaignState(string accountName, int campai
             IdCampaign: campaignId
         ));
 
-        if (campaignStateData == null || !campaignStateData.OwnCampaignExists)
-        {
-            return new NoExistCampaignState();
-        }
-
         // For information about Doppler's status code, check out here
         // https://github.com/MakingSense/Doppler/blob/develop/Doppler.Transversal/Classes/CampaignStatusEnum.cs
         var campaignStatus = campaignStateData.Status is DopplerCampaignStatusDraft or
             DopplerCampaignStatusABDraft ? CampaignStatus.Draft
             : campaignStateData.Status == DopplerCampaignStatusInWinnerInABSelectionProcess ? CampaignStatus.InWinnerInABSelectionProcess
             : CampaignStatus.Other;
 
+        if (campaignStateData == null || !campaignStateData.OwnCampaignExists)
+        {
+            return new NoExistCampaignState();
+        }
```

Comment:
Why did you move this validation down? In my opinion, we should return early when we know that the campaign does not exist. Am I missing something?

Reply:
I moved up the `campaignStatus` declaration variable because I use it in `NoTestABCampaignState()` method, and I thought that both validations should be together. Maybe I can separate them with the declaration between?